src/foam/nanos/crunch/Capability.js

```diff
@@ -392,7 +419,7 @@ foam.CLASS({
           subject.setUser((User) userDAO.find(ucj.getSourceId()));
         }
 
-        var prereqs = crunchService.getPrereqs(getId());
+        var prereqs = crunchService.getPrereqs(x.put("subject", ucj.getSubject(x)), getId());
```

Comment:
Thought, 
crunchService.getPrereqs(x, ucj, getId())  and the CrunchService can setup the subject.